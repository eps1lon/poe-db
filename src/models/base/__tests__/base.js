const { createOrm } = require('../../../db');
const base = require('../index');

const sequelize = createOrm({ logging: false });
const models = sequelize.models;
base(sequelize);

it('should apply the expansion without crash', () => {
  const model = sequelize.models.CraftingBenchOption;

  expect(model.denormalizeThrough).toBeDefined();
});

it('should denormalize extended props by joining the in a comma separated list', async () => {
  const model = models.CraftingBenchOption;

  const results = model.findAll({
    attributes: ['row'],
    limit: 3,
    include: [
      {
        model: models.BaseItemType,
        as: 'cost_base_item_types',
        attributes: ['row'],
        through: {
          attributes: ['value', 'priority'],
        },
      },
    ],
    order: [['row', 'asc']],
  });

  const denormalized = await model.denormalizeThrough(
    'cost_base_item_types',
    [
      { attribute: 'row', as: 'Cost_BaseItemTypesKeys', through_prop: false },
      { attribute: 'value', as: 'Cost_Values', through_prop: true },
    ],
    results,
  );

  expect(denormalized).toMatchSnapshot();
});

afterAll(() => {
  sequelize.close();
});
