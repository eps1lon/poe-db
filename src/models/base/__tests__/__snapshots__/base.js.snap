// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should denormalize extended props by joining the in a comma separated list 1`] = `
Array [
  Object {
    "Cost_BaseItemTypesKeys": "10",
    "Cost_Values": "1",
    "row": 0,
  },
  Object {
    "Cost_BaseItemTypesKeys": "9",
    "Cost_Values": "5",
    "row": 1,
  },
  Object {
    "Cost_BaseItemTypesKeys": "9",
    "Cost_Values": "10",
    "row": 2,
  },
]
`;
